### PR TITLE
new profile for checkers that pass

### PR DESCRIPTION
**Pull request by markro49**

### CODE REVIEW

File: src/main/java/org/apache/bcel/classfile/MethodParameter.java [hunk 1]
```diff
@@ -25,7 +25,6 @@
 
 import org.checkerframework.checker.index.qual.NonNegative;
 import org.checkerframework.checker.nullness.qual.Nullable;
-import org.checkerframework.checker.signedness.qual.Unsigned;
```
Comment (mernst) [inline src/main/java/org/apache/bcel/classfile/MethodParameter.java]: The changes in this file should be in a different pull request, so that each pull request is smaller and focused on a single concept.

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -32,5 +32,7 @@ jobs:
       with:
         distribution: 'temurin'
         java-version: ${{ matrix.java }}
+    - name: compile with passing checkers
+      run: mvn -V apache-rat:check -Prungood compile --file pom.xml
```
Comment (mernst) [inline .github/workflows/maven.yml]: `rungood` doesn't convey what this is about.  I would name it "suceedingTypecheckers" or "typecheckForbidErrors" or some better name than I am thinking of right now.

File: pom.xml [hunk 3]
```diff
@@ -78,6 +79,13 @@
     <commons.spotbugs.plugin.version>4.7.1.1</commons.spotbugs.plugin.version>
     <commons.spotbugs.impl.version>4.7.1</commons.spotbugs.impl.version>    
     <biz.aQute.bndlib.version>6.2.0</biz.aQute.bndlib.version>
+    <!-- default checkerframework annotation processors and compiler arguments -->
+    <ap1>org.checkerframework.checker.interning.InterningChecker</ap1>
```
Comment (mernst) [inline pom.xml]: `<ap>` is too short a mnemonic to be clear to a reader.  Even lengthening it to `<annoProc>` would be better.  I would go for `<annotationProcessor>` because it only appears here and the wordiness is not a problem.

**Review by mernst — APPROVED** [review]

File: src/main/java/org/apache/bcel/classfile/MethodParameter.java (hunk 1 above)
Comment (markro49) [reply src/main/java/org/apache/bcel/classfile/MethodParameter.java]: https://github.com/typetools/commons-bcel/pull/24
